chef/cookbooks/network/recipes/default.rb

```diff
@@ -393,6 +393,20 @@ def get_datapath_id_for_ovsbridge(bridge)
     nic.tx_offloading = node["network"]["enable_tx_offloading"] || true
   end
 
+  # if the nic is a vlan and the physical parent is not used, also set
+  # the mtu from the vlan nic for the parent
+  if nic.kind_of?(Nic::Vlan)
+    unless ifs[nic.parent].key? "mtu"
+      parent_nic = Nic.new(nic.parent)
+      # if there are multiple vlan's on the parent nic, use the highest mtu
+      if parent_nic.mtu < ifs[nic.name]["mtu"].to_i
+        Chef::Log.info("vlan #{nic.name} wants mtu #{nic.mtu} but parent #{parent_nic.name} " \
+                       "has no explicit mtu. Set mtu for #{parent_nic.name} to #{ifs[nic.name]['mtu']}")
```

Comment:
I think you need to make sure here, that ```parent_nic``` is not used by another network directly. It should be possible by looking at ```if_mappings```. It looks like this:

```
{
   "<networkname>" => [ "list", "of", "interfaces"],
   "<othernet>" => [ .... ]
}
```

As the list of interface always contains a complete chain that is used by the network e.g. [ eth0, eth0.300, br-public] you need to find all networks that contain ```nic.parent``` and have ```use_vlan != true```  (e.g. by looking at ```Barclamp::Inventory.get_network_by_type(node, <networkname>).use_vlan```

If there is such a device an its mtu is smailler that the mtu of the vlan we're currently looking at, I think we need to error out.